utils/rpms/packaging/Dockerfile.mockbuild

```diff
@@ -29,6 +29,21 @@ RUN dnf -y install mock make                                        \
                    python-srpm-macros rpmdevtools &&                \
     dnf -y clean all
 
+# Install bullseye coverage
+ARG JENKINS_URL
+ARG BULLSEYE_KEY
+RUN if [ "x$BULLSEYE" != "x" ]; then                                                \
+      dnf install xz && dnf clean all &&                                            \
+      curl ${JENKINS_URL}job/daos-stack/job/tools/job/master/lastSuccessfulBuild/artifact/bullseyecoverage-linux.tar \
+        --retry 10 --retry-max-time 60 --silent --show-error -o bullseye.tar.xz &&  \
+      mkdir -p bullseye &&                                                          \
+      tar -C bullseye --strip-components=1 -xf bullseye.tar.xz &&                   \
+      pushd bullseye &&                                                             \
+        ./install --quiet --key "${BULLSEYE_KEY}" --prefix /opt/BullseyeCoverage && \
+      popd &&                                                                       \
+      rm -rf bullseye.tar.xz bullseye;                                              \
+    fi
+
```

Comment:
I'm not sure this is going to have the effect that I suspect you are looking for.

I suspect you are wanting the RPMs to be built with the Bullseye compiler.  But this will not achieve that.  That is because you are installing the Bullseye compiler in the Docker image but the compiler that is used is the one that is installed in the **chroot** that `mock` creates to build the RPMs in.

I'm not sure that installing Bullseye, from a tarball into the chroot is going to be possible.  `mock` really wants everything it installs to be an RPM.  We could create a job that packages Bullseye up into an RPM so that `mock` can install it though.

Building RPMs with Bullseye isn't something we are doing in Jenkins either is it?  Given the above, I cannot imagine how we would be achieving that.

Reply:
From what I can tell with this change it does seem to be working. In https://github.com/daos-stack/daos/actions/runs/11186364681/job/31101247420?pr=15206 we end up going through this code path and installing bullseye:
```
#8 [ 4/11] RUN if [ "x$BULLSEYE" != "x" ]; then                                                      dnf install xz && dnf clean all &&                                                  curl https://build.hpdd.intel.com/job/daos-stack/job/tools/job/master/lastSuccessfulBuild/artifact/bullseyecoverage-linux.tar         --retry 10 --retry-max-time 60 --silent --show-error -o bullseye.tar.xz &&        mkdir -p bullseye &&                                                                tar -C bullseye --strip-components=1 -xf bullseye.tar.xz &&                         pushd bullseye &&                                                                     ./install --quiet --key "***" --prefix /opt/BullseyeCoverage &&       popd &&                                                                             rm -rf bullseye.tar.xz bullseye;                                                  fi
#8 DONE 14.6s
```

These RPMs are then uploaded to https://github.com/daos-stack/daos/actions/runs/11186364681/artifacts/2017684542 and used by the `Functional Testing (el8)` action, where we do see coverity files collected as part of the artifacts. (Note: the zip file of artifacts encounters an error when opening on windows, but works on linux).

Reply:
> From what I can tell with this change it does seem to be working. In [daos-stack/daos/actions/runs/11186364681/job/31101247420?pr=15206](https://github.com/daos-stack/daos/actions/runs/11186364681/job/31101247420?pr=15206) `Build RPM (el8)->Build RPM Docker Image` we end up going through this code path and installing bullseye:
> 
> ```
> #8 [ 4/11] RUN if [ "x$BULLSEYE" != "x" ]; then
>    dnf install xz && dnf clean all &&
>    curl https://build.hpdd.intel.com/job/daos-stack/job/tools/job/master/lastSuccessfulBuild/artifact/bullseyecoverage-linux.tar         --retry 10 --retry-max-time 60 --silent --show-error -o bullseye.tar.xz &&
>    mkdir -p bullseye &&
>    tar -C bullseye --strip-components=1 -xf bullseye.tar.xz &&
>    pushd bullseye &&
>    ./install --quiet --key "***" --prefix /opt/BullseyeCoverage &&
>    popd &&
>    rm -rf bullseye.tar.xz bullseye;
> fi
> #8 DONE 14.6s
> ```

Does it though?  I don't see the output from the `dnf install xz && dnf clean all` commands that I think we should see.  If if `xz` is already installed we should see at minimum something similar to:

```
dnf install xz
Last metadata expiration check: 1:05:35 ago on Tue 08 Oct 2024 08:24:46 AM EDT.
Package xz-1:5.4.6-3.fc40.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

My suggestion is to add a `set -x` to that step per my suggestion to see exactly what it's doing.

But even if Bullseye is being installed in the Docker image, I'm skeptical that is what is actually being used to build the RPMs.  Indeed!  If we look at the root.log in https://github.com/daos-stack/daos/actions/runs/11186364681/artifacts/2017683891 we can see that bullseye is actually installed from an RPM:

```
DEBUG util.py:558:  Executing command: ['/usr/bin/dnf-3', 'builddep', '--installroot', '/var/lib/mock/rocky+epel-8-x86_64/root/', '--releasever', '8.6', '--disablerepo', '*', '--enablerepo', '*-artifactory', '--disablerepo', '*-debug*', '--disablerepo', 'sclo*', '--setopt=deltarpm=False', '--setopt=allow_vendor_change=yes', '--allowerasing', '--disableplugin=local', '--disableplugin=spacewalk', '--disableplugin=versionlock', '/var/lib/mock/rocky+epel-8-x86_64/root//builddir/build/SRPMS/daos-2.7.100-5.1.g20d01303.el8.src.rpm'] with env {'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOME': '/var/lib/mock/rocky+epel-8-x86_64/root/installation-homedir', 'HOSTNAME': 'mock', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'LANG': 'C.UTF-8', 'CCACHE_DIR': '/var/tmp/ccache', 'CCACHE_UMASK': '002', 'LC_MESSAGES': 'C.UTF-8'} and shell False
DEBUG util.py:461:  No matches found for the following disable plugin patterns: local, spacewalk, versionlock
DEBUG util.py:461:  No repository match: sclo*
DEBUG util.py:463:  daos_ci-rocky8-base-artifactory                  70 kB/s | 1.4 kB     00:00    
DEBUG util.py:463:  daos_ci-rocky8-extras-artifactory                58 kB/s | 1.4 kB     00:00    
DEBUG util.py:463:  daos_ci-rocky8-appstream-artifactory             76 kB/s | 1.8 kB     00:00    
DEBUG util.py:463:  daos_ci-rocky8-powertools-artifactory           129 kB/s | 4.8 kB     00:00    
DEBUG util.py:463:  daos_ci-el8-epel-artifactory                    165 kB/s | 4.4 kB     00:00    
DEBUG util.py:463:  daos_ci-el8-epel-modular-artifactory            106 kB/s | 3.0 kB     00:00    
DEBUG util.py:463:  daos-stack-deps-el-8-x86_64-stable-local-artifa 4.7 kB/s | 475  B     00:00    
DEBUG util.py:463:  daos-stack-daos-el-8-x86_64-stable-local-artifa 6.6 kB/s | 475  B     00:00    
DEBUG util.py:463:  daos_ci-intel-tools-el-8-x86_64-stable-local-ar 7.7 kB/s | 475  B     00:00    
DEBUG util.py:463:  Package gcc-c++-8.5.0-10.1.el8_6.x86_64 is already installed.
DEBUG util.py:463:  Dependencies resolved.
DEBUG util.py:463:  =====================================================================================================================================
DEBUG util.py:463:   Package                     Arch    Version                              Repository                                             Size
DEBUG util.py:463:  =====================================================================================================================================
DEBUG util.py:463:  Installing:
…
DEBUG util.py:463:   bullseye                    x86_64  9.1.1-1.el8                          daos-stack-deps-el-8-x86_64-stable-local-artifactory  2.2 M
…
```

I'm actually quite surprised by this.  Indeed, this is a result of https://github.com/daos-stack/bullseye/ and https://github.com/daos-stack/daos/blob/9211793ce45f6b938a71b0b7f6891059d15330f2/utils/rpms/daos.spec#L40-L42

What I'm unclear about is what triggered the build and test with Bullseye?  That's normally a commit pragma or in Jenkins, a default-false build parameter isn't it?